src/os/mod.rs

```diff
@@ -22,20 +20,57 @@ pub fn force_kill(pid: u32) -> bool {
 }
 
 /// Gracefully kill process.
-///
 /// Results in undefined behavior if PID is invalid.
 ///
 /// # Panics
-///
 /// Panics on platforms other than Unix.
 #[allow(unreachable_code, dead_code, unused_variables)]
 pub fn kill_gracefully(pid: u32) -> bool {
     #[cfg(unix)]
-    unsafe {
-        return unix::kill_gracefully(pid);
-    }
+    return unix_signal(pid, signal::SIGTERM);
 
     unimplemented!(
         "gracefully killing Minecraft server process not implemented on non-Unix platforms"
     );
 }
+
+/// Freeze process.
+/// Results in undefined behavior if PID is invaild.
+///
+/// # Panics
+/// Panics on platforms other than Unix.
+#[allow(unreachable_code)]
+pub fn freeze(pid: u32) -> bool {
+    #[cfg(unix)]
+    return unix_signal(pid, signal::SIGSTOP);
+
+    unimplemented!(
+        "Freezing the Minecraft server process is not implemented on non-Unix platforms."
```

Comment:
Nitpick:

I use all lowercase messages without a full stop everywhere else, maybe we should do that here too.

```diff
-        "Freezing the Minecraft server process is not implemented on non-Unix platforms."
+        "freezing the Minecraft server process is not implemented on non-Unix platforms"
```